src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java

```diff
@@ -296,7 +315,50 @@ private void loadPathToMainScript() {
         this.appiumJS = findMainScript();
     }
 
+    private String parseCapabilitiesIfWindows() {
+        StringBuilder result = new StringBuilder();
+        final List<String> pathCapabilities = ImmutableList.of(AndroidMobileCapabilityType.KEYSTORE_PATH,
+                AndroidMobileCapabilityType.CHROMEDRIVER_EXECUTABLE, MobileCapabilityType.APP);
+
+        if (capabilities != null) {
+            Map<String, ?> capabilitiesMap = capabilities.asMap();
+            Set<? extends Map.Entry<String, ?>> entries = capabilitiesMap.entrySet();
+
+            for (Map.Entry<String, ?> entry : entries) {
+                Object value = entry.getValue();
+
+                if (value == null) {
+                    continue;
+                }
+
+                if (value instanceof String) {
+                    String valueString = (String) value;
+                    if (pathCapabilities.contains(entry.getKey())) {
+                        value = "\\\"" + valueString.replace("\\", "/") + "\\\"";
+                    } else {
+                        value = "\\\"" + valueString + "\\\"";
+                    }
+                } else {
+                    value = String.valueOf(value);
+                }
+
+                String key = "\\\"" + entry.getKey() + "\\\"";
+                if (result.length() > 0) {
+                    result.append(", ");
+                }
+                result.append(key);
```

Comment:
this method returns self instance, so all calls could be changed like result.append().append()....

Reply:
you're right looks cleaner...pushed